nixos/modules/config/xdg/portals/termfilechooser.nix

```diff
@@ -0,0 +1,79 @@
+{ config
+, pkgs
+, lib
+, ...
+}:
+let
+  cfg = config.xdg.portal.termfilechooser;
+  package = pkgs.xdg-desktop-portal-termfilechooser;
+  settingsFormat = pkgs.formats.ini { };
+  configFile = settingsFormat.generate "xdg-desktop-portal-termfilechooser.ini" cfg.settings;
+in
+{
+  meta = {
+    maintainers = [ lib.maintainers.soispha ];
+  };
+
+  options.xdg.portal.termfilechooser = {
+    enable = lib.mkEnableOption (lib.mdDoc ''
+      Desktop portal for wlroots-based desktops
```

Comment:
Why limit it to wlroots?

Reply:
Because it's [based on `xdg-desktop-portal-wlr`](https://github.com/GermainZ/xdg-desktop-portal-termfilechooser/blob/main/README.md?plain=1#L6). It could work on non-wlr desktops, but I have no way to test this.

Reply:
Although it [seems](https://github.com/GermainZ/xdg-desktop-portal-termfilechooser/blob/main/termfilechooser.portal#L4) to support non-wlr desktops.

Reply:
`UseIn` is deprecated. Nowadays, user can configure which portal to use. Since the portal is supposed to allow running any command, I do not see why it would not work everywhere.

I understood the x-d-p-wlr reference only as an acknowledgement/credits.

Reply:
This seems perfectly logical. I'll change the enable string